File: Easy/9 - PalindromeNumber/solution.ts
/**
 * Question 9. Palindrome Number (https://leetcode-cn.com/problems/palindrome-number/)
 *
 * 执行用时: 204 ms, 在所有 typescript 提交中击败了 85.00% 的用户
 * 内存消耗: 44.1 MB, 在所有 typescript 提交中击败了 100.00% 的用户
 *
 * @param {number} x
 * @returns {boolean}
 */
function isPalindrome(x: number): boolean {
  let xString: string = `${x}`;
  let leftPointer: number = 0;
  let rightPointer: number = xString.length - 1;

  while (leftPointer < rightPointer) {
    if (xString[leftPointer++] !== xString[rightPointer--]) {
      return false;
    }
  }

  return true;
}

File: Easy/9 - PalindromeNumber/Solution.cs
/// Question 9. Palindrome Number (https://leetcode-cn.com/problems/palindrome-number/)

public class Solution
{
    // 执行用时 : 76 ms , 在所有 C# 提交中击败了 85.07% 的用户
    // 内存消耗 : 16.5 MB, 在所有 C# 提交中击败了 40.00% 的用户

    public bool IsPalindrome(int x)
    {
        string xInString = x.ToString();

        int leftPointer = 0;
        int rightPointer = xInString.Length - 1;

        while (leftPointer < rightPointer)
        {
            if (xInString[leftPointer++] != xInString[rightPointer--])
            {
                return false;
            }
        }

        return true;
    }
}